include/RAJA/pattern/reduce.hpp

```diff
@@ -64,6 +65,67 @@
 namespace RAJA
 {
 
+namespace reduce
```

Comment:
It's also in a common location so other reducers could be restructured to adhere to a similar design. (and leverage OpenMP locks -- something that @trws would certainly prefer as well)